app/renderer/js/spellchecker.js

```diff
@@ -26,13 +26,13 @@ class SetupSpellChecker {
 		}
 	}
 
-	enableContextMenu() {
+	enableContextMenu(serverLanguage) {
 		if (this.SpellCheckHandler) {
 			this.SpellCheckHandler.attachToInput();
-
-			const userLanguage = ConfigUtil.getConfigItem('spellcheckerLanguage');
-
-			this.SpellCheckHandler.switchLanguage(userLanguage);
+			this.SpellCheckHandler.switchLanguage(serverLanguage);
+			this.SpellCheckHandler.currentSpellcheckerChanged.subscribe(() => {
```

Comment:
@kanishk98  Can you help me understand what this does? I need to know this for BrowserView. Getting problems due to this.

Reply:
Sure. 

This basically tells the spellchecker module to keep track of changes in the input text and if the total body of text is in a different language than the one the module assumed in the beginning, then it applies spellcheck according to the new language.

Reply:
I think it'll be better for you if you check out the code of this class, but I doubt if the module itself is causing problems. 

Can you tell me what's causing trouble? Maybe I can help.